diff --git a/composer.json b/composer.json
index 086f4c5..3b2ea53 100644
--- a/composer.json
+++ b/composer.json
@@ -24,6 +24,6 @@
         }
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8"
+        "phpunit/phpunit": "^4.8.35|^5|^6|^7"
     }
 }
diff --git a/tests/BaconQrCode/Common/BitArrayTest.php b/tests/BaconQrCode/Common/BitArrayTest.php
index 81bcbce..fe154f5 100644
--- a/tests/BaconQrCode/Common/BitArrayTest.php
+++ b/tests/BaconQrCode/Common/BitArrayTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class BitArrayTest extends TestCase
 {
@@ -198,4 +198,4 @@ class BitArrayTest extends TestCase
         $this->assertTrue($array->isRange(0, 64, true));
         $this->assertFalse($array->isRange(0, 64, false));
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/BitMatrixTest.php b/tests/BaconQrCode/Common/BitMatrixTest.php
index 89a5881..80ab962 100644
--- a/tests/BaconQrCode/Common/BitMatrixTest.php
+++ b/tests/BaconQrCode/Common/BitMatrixTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class BitMatrixTest extends TestCase
 {
@@ -116,4 +116,4 @@ class BitMatrixTest extends TestCase
             $this->assertEquals($on, $array3->get($x));
         }
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/BitUtilsTest.php b/tests/BaconQrCode/Common/BitUtilsTest.php
index b80ff7d..47ea610 100644
--- a/tests/BaconQrCode/Common/BitUtilsTest.php
+++ b/tests/BaconQrCode/Common/BitUtilsTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class BitUtilsTest extends TestCase
 {
@@ -27,4 +27,4 @@ class BitUtilsTest extends TestCase
         $this->assertEquals(0, BitUtils::numberOfTrailingZeros(15));
         $this->assertEquals(2, BitUtils::numberOfTrailingZeros(20));
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/ErrorCorrectionLevelTest.php b/tests/BaconQrCode/Common/ErrorCorrectionLevelTest.php
index 736e995..7ec63da 100644
--- a/tests/BaconQrCode/Common/ErrorCorrectionLevelTest.php
+++ b/tests/BaconQrCode/Common/ErrorCorrectionLevelTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class ErrorCorrectionLevelTest extends TestCase
 {
@@ -29,12 +29,12 @@ class ErrorCorrectionLevelTest extends TestCase
         $this->assertEquals(0x3, ErrorCorrectionLevel::Q);
     }
 
+    /**
+     * @expectedException BaconQrCode\Exception\UnexpectedValueException
+     * @expectedExceptionMessage Value not a const in enum BaconQrCode\Common\ErrorCorrectionLevel
+     */
     public function testInvalidErrorCorrectionLevelThrowsException()
     {
-        $this->setExpectedException(
-            'BaconQrCode\Exception\UnexpectedValueException',
-            'Value not a const in enum BaconQrCode\Common\ErrorCorrectionLevel'
-        );
         new ErrorCorrectionLevel(4);
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/FormatInformationTest.php b/tests/BaconQrCode/Common/FormatInformationTest.php
index 5f6ee58..b2f115c 100644
--- a/tests/BaconQrCode/Common/FormatInformationTest.php
+++ b/tests/BaconQrCode/Common/FormatInformationTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class FormatInformationTest extends TestCase
 {
@@ -101,4 +101,4 @@ class FormatInformationTest extends TestCase
             )
         );
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/ModeTest.php b/tests/BaconQrCode/Common/ModeTest.php
index 4daab7c..4eff32d 100644
--- a/tests/BaconQrCode/Common/ModeTest.php
+++ b/tests/BaconQrCode/Common/ModeTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class ModeTest extends TestCase
 {
@@ -31,12 +31,12 @@ class ModeTest extends TestCase
         $this->assertEquals(0x8, Mode::KANJI);
     }
 
+    /**
+     * @expectedException BaconQrCode\Exception\UnexpectedValueException
+     * @expectedExceptionMessage Value not a const in enum BaconQrCode\Common\Mode
+     */
     public function testInvalidModeThrowsException()
     {
-        $this->setExpectedException(
-            'BaconQrCode\Exception\UnexpectedValueException',
-            'Value not a const in enum BaconQrCode\Common\Mode'
-        );
         new Mode(10);
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/ReedSolomonCodecTest.php b/tests/BaconQrCode/Common/ReedSolomonCodecTest.php
index 604641a..2de001e 100644
--- a/tests/BaconQrCode/Common/ReedSolomonCodecTest.php
+++ b/tests/BaconQrCode/Common/ReedSolomonCodecTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 use SplFixedArray;
 
 class ReedSolomonTest extends TestCase
@@ -108,4 +108,4 @@ class ReedSolomonTest extends TestCase
             $this->assertEquals($block, $tBlock, 'Decoder did not correct errors');
         }
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Common/VersionTest.php b/tests/BaconQrCode/Common/VersionTest.php
index 8b3fc01..f16576a 100644
--- a/tests/BaconQrCode/Common/VersionTest.php
+++ b/tests/BaconQrCode/Common/VersionTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Common;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class VersionTest extends TestCase
 {
@@ -85,4 +85,4 @@ class VersionTest extends TestCase
         $this->assertNotNull($version);
         $this->assertEquals($expectedVersion, $version->getVersionNumber());
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Encoder/EncoderTest.php b/tests/BaconQrCode/Encoder/EncoderTest.php
index 31cdaa4..8f67fa3 100644
--- a/tests/BaconQrCode/Encoder/EncoderTest.php
+++ b/tests/BaconQrCode/Encoder/EncoderTest.php
@@ -13,7 +13,7 @@ use BaconQrCode\Common\BitArray;
 use BaconQrCode\Common\ErrorCorrectionLevel;
 use BaconQrCode\Common\Mode;
 use BaconQrCode\Common\Version;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 use ReflectionClass;
 use ReflectionMethod;
 use SplFixedArray;
@@ -212,6 +212,10 @@ class EncoderTest extends TestCase
         $this->assertEquals(' ..X..... ....', $bits->__toString());
     }
 
+    /**
+     * @expectedException BaconQrCode\Exception\WriterException
+     * @expectedExceptionMessage Invalid alphanumeric code
+     */
     public function testAppendBytes()
     {
         // Should use appendNumericBytes.
@@ -263,10 +267,6 @@ class EncoderTest extends TestCase
         $this->assertEquals(' .XX.XX.. XXXXX', $bits->__toString());
 
         // Lower letters such as 'a' cannot be encoded in alphanumeric mode.
-        $this->setExpectedException(
-            'BaconQrCode\Exception\WriterException',
-            'Invalid alphanumeric code'
-        );
         $this->methods['appendBytes']->invoke(
             null,
             "a",
@@ -396,6 +396,10 @@ class EncoderTest extends TestCase
         $this->assertEquals('', $bits->__toString());
     }
 
+    /**
+     * @expectedException BaconQrCode\Exception\WriterException
+     * @expectedExceptionMessage Invalid alphanumeric code
+     */
     public function testAppendAlphanumericBytes()
     {
         $bits = new BitArray();
@@ -416,7 +420,6 @@ class EncoderTest extends TestCase
         $this->assertEquals('', $bits->__toString());
 
         // Invalid data
-        $this->setExpectedException('BaconQrCode\Exception\WriterException', 'Invalid alphanumeric code');
         $bits = new BitArray();
         $this->methods['appendAlphanumericBytes']->invoke(null, 'abc', $bits);
     }
@@ -465,4 +468,4 @@ class EncoderTest extends TestCase
         $expected  = SplFixedArray::fromArray(array(0, 3, 130, 179, 194, 0, 55, 211, 110, 79, 98, 72, 170, 96, 211, 137, 213), false);
         $this->assertEquals($expected, $ecBytes);
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Encoder/MaskUtilTest.php b/tests/BaconQrCode/Encoder/MaskUtilTest.php
index a5c3865..6ffef9b 100644
--- a/tests/BaconQrCode/Encoder/MaskUtilTest.php
+++ b/tests/BaconQrCode/Encoder/MaskUtilTest.php
@@ -9,7 +9,7 @@
 
 namespace BaconQrCode\Encoder;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class MaskUtilTest extends TestCase
 {
@@ -278,4 +278,4 @@ class MaskUtilTest extends TestCase
         $matrix->set(5, 0, 0);
         $this->assertEquals(30, MaskUtil::applyMaskPenaltyRule4($matrix));
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Encoder/MatrixUtilTest.php b/tests/BaconQrCode/Encoder/MatrixUtilTest.php
index bf3544f..7a82a47 100644
--- a/tests/BaconQrCode/Encoder/MatrixUtilTest.php
+++ b/tests/BaconQrCode/Encoder/MatrixUtilTest.php
@@ -12,7 +12,7 @@ namespace BaconQrCode\Encoder;
 use BaconQrCode\Common\BitArray;
 use BaconQrCode\Common\ErrorCorrectionLevel;
 use BaconQrCode\Common\Version;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 use ReflectionClass;
 use ReflectionMethod;
 
@@ -333,4 +333,4 @@ class MatrixUtilTest extends TestCase
         $this->methods['makeTypeInfoBits']->invoke(null, new ErrorCorrectionLevel(ErrorCorrectionLevel::M), 5, $bits);
         $this->assertEquals(' X......X X..XXX.', $bits->__toString());
     }
-}
\ No newline at end of file
+}
diff --git a/tests/BaconQrCode/Renderer/Text/HtmlTest.php b/tests/BaconQrCode/Renderer/Text/HtmlTest.php
index 0c69dd2..9324ff9 100644
--- a/tests/BaconQrCode/Renderer/Text/HtmlTest.php
+++ b/tests/BaconQrCode/Renderer/Text/HtmlTest.php
@@ -12,7 +12,7 @@ namespace BaconQrCode\Encoder;
 use BaconQrCode\Common\ErrorCorrectionLevel;
 use BaconQrCode\Renderer\Text\Html;
 use BaconQrCode\Writer;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class HtmlTest extends TestCase
 {
diff --git a/tests/BaconQrCode/Renderer/Text/TextTest.php b/tests/BaconQrCode/Renderer/Text/TextTest.php
index d94e8e5..469e0ca 100644
--- a/tests/BaconQrCode/Renderer/Text/TextTest.php
+++ b/tests/BaconQrCode/Renderer/Text/TextTest.php
@@ -12,7 +12,7 @@ namespace BaconQrCode\Encoder;
 use BaconQrCode\Common\ErrorCorrectionLevel;
 use BaconQrCode\Renderer\Text\Plain;
 use BaconQrCode\Writer;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase;
 
 class PlainTest extends TestCase
 {
