diff --git a/bin/certificate-info.php b/bin/certificate-info.php
index 61f6dab..9c94503 100644
--- a/bin/certificate-info.php
+++ b/bin/certificate-info.php
@@ -1,3 +1,4 @@
+#!/usr/bin/env php
 <?php
 
 /*
@@ -7,8 +8,8 @@
  * SPDX-License-Identifier: AGPL-3.0+
  */
 
-require_once dirname(__DIR__).'/vendor/autoload.php';
-$baseDir = dirname(__DIR__);
+require_once '/usr/share/php/LC/Node/autoload.php';
+$baseDir = '/usr/share/vpn-server-node';
 
 use LC\Common\FileIO;
 
diff --git a/bin/generate-firewall.php b/bin/generate-firewall.php
index 9785cfc..af451bd 100644
--- a/bin/generate-firewall.php
+++ b/bin/generate-firewall.php
@@ -1,3 +1,4 @@
+#!/usr/bin/env php
 <?php
 
 /*
@@ -7,8 +8,8 @@
  * SPDX-License-Identifier: AGPL-3.0+
  */
 
-require_once dirname(__DIR__).'/vendor/autoload.php';
-$baseDir = dirname(__DIR__);
+require_once '/usr/share/php/LC/Node/autoload.php';
+$baseDir = '/usr/share/vpn-server-node';
 
 use LC\Common\Config;
 use LC\Common\FileIO;
diff --git a/bin/server-config.php b/bin/server-config.php
index 1a59896..5cd6c37 100644
--- a/bin/server-config.php
+++ b/bin/server-config.php
@@ -1,3 +1,4 @@
+#!/usr/bin/env php
 <?php
 
 /*
@@ -7,8 +8,8 @@
  * SPDX-License-Identifier: AGPL-3.0+
  */
 
-require_once dirname(__DIR__).'/vendor/autoload.php';
-$baseDir = dirname(__DIR__);
+require_once '/usr/share/php/LC/Node/autoload.php';
+$baseDir = '/usr/share/vpn-server-node';
 
 use LC\Common\Config;
 use LC\Common\HttpClient\CurlHttpClient;
diff --git a/config/config.php.example b/config/config.php.example
index 0112e14..c5ecff0 100644
--- a/config/config.php.example
+++ b/config/config.php.example
@@ -3,8 +3,8 @@
 return [
     // Override the user/group under which OpenVPN will run, on CentOS/Fedora this
     // is openvpn:openvpn, on Debian this is nobody:nogroup
-    'vpnUser' => 'openvpn',
-    'vpnGroup' => 'openvpn',
+    'vpnUser' => 'nobody',
+    'vpnGroup' => 'nogroup',
 
     'apiUser' => 'vpn-server-node',
     'apiPass' => 'XXX-vpn-server-node/vpn-server-api-XXX',
diff --git a/libexec/client-connect.php b/libexec/client-connect.php
index b4cffde..0f6b13a 100644
--- a/libexec/client-connect.php
+++ b/libexec/client-connect.php
@@ -1,3 +1,4 @@
+#!/usr/bin/env php
 <?php
 
 /*
@@ -7,8 +8,8 @@
  * SPDX-License-Identifier: AGPL-3.0+
  */
 
-require_once dirname(__DIR__).'/vendor/autoload.php';
-$baseDir = dirname(__DIR__);
+require_once '/usr/share/php/LC/Node/autoload.php';
+$baseDir = '/usr/share/vpn-server-node';
 
 use LC\Common\Config;
 use LC\Common\HttpClient\CurlHttpClient;
diff --git a/libexec/client-disconnect.php b/libexec/client-disconnect.php
index a7c75b3..0377784 100644
--- a/libexec/client-disconnect.php
+++ b/libexec/client-disconnect.php
@@ -1,3 +1,4 @@
+#!/usr/bin/env php
 <?php
 
 /*
@@ -7,8 +8,8 @@
  * SPDX-License-Identifier: AGPL-3.0+
  */
 
-require_once dirname(__DIR__).'/vendor/autoload.php';
-$baseDir = dirname(__DIR__);
+require_once '/usr/share/php/LC/Node/autoload.php';
+$baseDir = '/usr/share/vpn-server-node';
 
 use LC\Common\Config;
 use LC\Common\HttpClient\CurlHttpClient;
diff --git a/src/OpenVpn.php b/src/OpenVpn.php
index 61d9cad..880ac86 100644
--- a/src/OpenVpn.php
+++ b/src/OpenVpn.php
@@ -18,8 +18,8 @@ use RuntimeException;
 
 class OpenVpn
 {
-    // CentOS
-    const LIBEXEC_DIR = '/usr/libexec/vpn-server-node';
+    // Debian
+    const LIBEXEC_DIR = '/usr/lib/vpn-server-node';
 
     /** @var string */
     private $vpnConfigDir;
diff --git a/tests/data/internet-0.conf b/tests/data/internet-0.conf
index 0bdc4ac..03897d3 100644
--- a/tests/data/internet-0.conf
+++ b/tests/data/internet-0.conf
@@ -9,8 +9,8 @@ auth none
 ca tls/internet/ca.crt
 cert tls/internet/server.crt
 cipher AES-256-GCM
-client-connect /usr/libexec/vpn-server-node/client-connect
-client-disconnect /usr/libexec/vpn-server-node/client-disconnect
+client-connect /usr/lib/vpn-server-node/client-connect
+client-disconnect /usr/lib/vpn-server-node/client-disconnect
 dev tun-1-0
 dev-type tun
 dh none
