diff --git a/config/config.php.example b/config/config.php.example
index 0112e14..c5ecff0 100644
--- a/config/config.php.example
+++ b/config/config.php.example
@@ -3,8 +3,8 @@
 return [
     // Override the user/group under which OpenVPN will run, on CentOS/Fedora this
     // is openvpn:openvpn, on Debian this is nobody:nogroup
-    'vpnUser' => 'openvpn',
-    'vpnGroup' => 'openvpn',
+    'vpnUser' => 'nobody',
+    'vpnGroup' => 'nogroup',
 
     'apiUser' => 'vpn-server-node',
     'apiPass' => 'XXX-vpn-server-node/vpn-server-api-XXX',
diff --git a/src/OpenVpn.php b/src/OpenVpn.php
index c168e53..4cb85f8 100644
--- a/src/OpenVpn.php
+++ b/src/OpenVpn.php
@@ -19,8 +19,8 @@ use RuntimeException;
 
 class OpenVpn
 {
-    // CentOS
-    const LIBEXEC_DIR = '/usr/libexec/vpn-server-node';
+    // Debian
+    const LIBEXEC_DIR = '/usr/lib/vpn-server-node';
 
     /** @var string */
     private $vpnConfigDir;
diff --git a/tests/data/internet-0.conf b/tests/data/internet-0.conf
index ea18962..cf1b94b 100644
--- a/tests/data/internet-0.conf
+++ b/tests/data/internet-0.conf
@@ -6,8 +6,8 @@
 # ******************************************
 #
 cipher AES-256-GCM
-client-connect /usr/libexec/vpn-server-node/client-connect
-client-disconnect /usr/libexec/vpn-server-node/client-disconnect
+client-connect /usr/lib/vpn-server-node/client-connect
+client-disconnect /usr/lib/vpn-server-node/client-disconnect
 dev tun0
 dev-type tun
 dh none
