diff --git a/vpn-daemon/main.go b/vpn-daemon/main.go
index 1c4020a..1598cfe 100644
--- a/vpn-daemon/main.go
+++ b/vpn-daemon/main.go
@@ -41,9 +41,9 @@ import (
 
 // the full path to the CA, certificate and key files
 var (
-	tlsCaPath   = "ca.crt"
-	tlsCertPath = "server.crt"
-	tlsKeyPath  = "server.key"
+	tlsCaPath   = "/etc/ssl/certs/vpn-daemon-ca.pem"
+	tlsCertPath = "/etc/ssl/certs/vpn-daemon.pem"
+	tlsKeyPath  = "/etc/ssl/private/vpn-daemon.pem"
 )
 
 type vpnClientInfo struct {
