diff --git a/vpn-daemon/main.go b/vpn-daemon/main.go
index 1c4020a..bebf0ec 100644
--- a/vpn-daemon/main.go
+++ b/vpn-daemon/main.go
@@ -41,9 +41,9 @@ import (
 
 // the full path to the CA, certificate and key files
 var (
-	tlsCaPath   = "ca.crt"
-	tlsCertPath = "server.crt"
-	tlsKeyPath  = "server.key"
+	tlsCaPath   = "/etc/ssl/vpn-daemon/ca.crt"
+	tlsCertPath = "/etc/ssl/vpn-daemon/server.crt"
+	tlsKeyPath  = "/etc/ssl/vpn-daemon/private/server.key"
 )
 
 type vpnClientInfo struct {
