diff --git a/src/MetadataSource.php b/src/MetadataSource.php
index b434097..58ccd04 100644
--- a/src/MetadataSource.php
+++ b/src/MetadataSource.php
@@ -158,7 +158,7 @@ class MetadataSource implements IdpSourceInterface
     private function importMetadata(HttpClientInterface $httpClient, $metadataUrl, array $publicKeyFileList, $forceDownload)
     {
         $this->logger->notice(\sprintf('[%s] attempting to update metadata', $metadataUrl));
-        $metadataFile = \sprintf('%s/%s.xml', $this->dynamicDir, Base64UrlSafe::encodeUnpadded($metadataUrl));
+        $metadataFile = \sprintf('%s/%s.xml', $this->dynamicDir, \rtrim(Base64UrlSafe::encode($metadataUrl), '='));
         $requestHeaders = [];
         if (!$forceDownload) {
             if (false !== $metadataString = @\file_get_contents($metadataFile)) {
