diff --git a/bin/update-metadata.php b/bin/update-metadata.php
index af591b7..d4d6250 100644
--- a/bin/update-metadata.php
+++ b/bin/update-metadata.php
@@ -22,14 +22,14 @@
  * SOFTWARE.
  */
 
-require_once \dirname(__DIR__).'/vendor/autoload.php';
+require_once '/usr/share/php/fkooman/SAML/SP/autoload.php';
 
 use fkooman\SAML\SP\CurlHttpClient;
 use fkooman\SAML\SP\Log\SyslogLogger;
 use fkooman\SAML\SP\MetadataSource;
 use fkooman\SAML\SP\Web\Config;
 
-$baseDir = \dirname(__DIR__);
+$baseDir = '/usr/share/php-saml-sp';
 $logger = new SyslogLogger(\basename($argv[0]));
 
 try {
diff --git a/bin/validate-metadata.php b/bin/validate-metadata.php
index 37e07f3..301a966 100644
--- a/bin/validate-metadata.php
+++ b/bin/validate-metadata.php
@@ -22,7 +22,7 @@
  * SOFTWARE.
  */
 
-require_once \dirname(__DIR__).'/vendor/autoload.php';
+require_once '/usr/share/php/fkooman/SAML/SP/autoload.php';
 
 use fkooman\SAML\SP\Crypto;
 use fkooman\SAML\SP\PublicKey;
diff --git a/src/Api/SamlAuth.php b/src/Api/SamlAuth.php
index de9f587..eac5994 100644
--- a/src/Api/SamlAuth.php
+++ b/src/Api/SamlAuth.php
@@ -48,7 +48,7 @@ class SamlAuth
 
     public function __construct()
     {
-        $this->config = Config::fromFile(\dirname(\dirname(__DIR__)).'/config/config.php');
+        $this->config = Config::fromFile('/usr/share/php-saml-sp/config/config.php');
         $this->request = new Request($_SERVER, $_GET, $_POST);
         $this->session = new SeSession($this->config->getSecureCookie());
         $this->dateTime = new DateTime();
diff --git a/web/index.php b/web/index.php
index c988e71..d3202ae 100644
--- a/web/index.php
+++ b/web/index.php
@@ -22,7 +22,7 @@
  * SOFTWARE.
  */
 
-require_once \dirname(__DIR__).'/vendor/autoload.php';
+require_once '/usr/share/php/fkooman/SAML/SP/autoload.php';
 
 use fkooman\SAML\SP\CryptoKeys;
 use fkooman\SAML\SP\Log\SyslogLogger;
@@ -37,7 +37,7 @@ use fkooman\SAML\SP\Web\SeCookie;
 use fkooman\SAML\SP\Web\Service;
 use fkooman\SAML\SP\Web\Tpl;
 
-$baseDir = \dirname(__DIR__);
+$baseDir = '/usr/share/php-saml-sp';
 $dataDir = $baseDir.'/data';
 $logger = new SyslogLogger('php-saml-sp');
 
@@ -76,7 +76,7 @@ try {
     $spInfo = new SpInfo(
         $spEntityId,
         // AuthnRequest / LogoutRequest / Decryption <EncryptedAssertion>
-        CryptoKeys::load($baseDir.'/keys'),
+        CryptoKeys::load($baseDir.'/keys/private', $baseDir.'/keys'),
         $request->getRootUri().'acs',
         $config->getRequireEncryption(),
         $config->getServiceNames()
