diff --git a/src/Api/SamlAuth.php b/src/Api/SamlAuth.php
index 04fa0b9..5c5c0bb 100644
--- a/src/Api/SamlAuth.php
+++ b/src/Api/SamlAuth.php
@@ -48,7 +48,7 @@ class SamlAuth
 
     public function __construct()
     {
-        $this->config = Config::fromFile(\dirname(\dirname(__DIR__)).'/config/config.php');
+        $this->config = Config::fromFile('/usr/share/php-saml-sp/config/config.php');
         $this->request = new Request($_SERVER, $_GET, $_POST);
         $this->session = new SeSession($this->config->getSecureCookie());
         $this->dateTime = new DateTime();
diff --git a/web/index.php b/web/index.php
index b81ddc7..3e8878b 100644
--- a/web/index.php
+++ b/web/index.php
@@ -22,7 +22,7 @@
  * SOFTWARE.
  */
 
-require_once \dirname(__DIR__).'/vendor/autoload.php';
+require_once '/usr/share/php/fkooman/SAML/SP/autoload.php';
 
 use fkooman\SAML\SP\CryptoKeys;
 use fkooman\SAML\SP\SeSession;
@@ -36,7 +36,7 @@ use fkooman\SAML\SP\Web\Service;
 use fkooman\SAML\SP\Web\Tpl;
 use fkooman\SAML\SP\XmlIdpInfoSource;
 
-$baseDir = \dirname(__DIR__);
+$baseDir = '/usr/share/php-saml-sp';
 
 try {
     $config = Config::fromFile($baseDir.'/config/config.php');
@@ -72,7 +72,7 @@ try {
     $spInfo = new SpInfo(
         $spEntityId,
         // AuthnRequest / LogoutRequest / Decryption <EncryptedAssertion>
-        CryptoKeys::load($baseDir.'/keys'),
+        CryptoKeys::load($baseDir.'/keys/private', $baseDir.'/keys'),
         $request->getRootUri().'acs',
         $config->getRequireEncryption(),
         $config->getServiceNames()
