Index: openvpn-plugin-auth-script-0.0~git02e08ef/meson.build
===================================================================
--- openvpn-plugin-auth-script-0.0~git02e08ef.orig/meson.build
+++ openvpn-plugin-auth-script-0.0~git02e08ef/meson.build
@@ -12,5 +12,7 @@ library(
     c_args : [
         '-I/usr/include/openvpn', # Debian
         '-I/usr/local/include'    # FreeBSD
-    ]
+    ],
+    install : true,
+    install_dir : 'lib/openvpn'
 )
