diff --git a/src/OAuth/PublicSigner.php b/src/OAuth/PublicSigner.php
index e31f50c..a875390 100644
--- a/src/OAuth/PublicSigner.php
+++ b/src/OAuth/PublicSigner.php
@@ -41,13 +41,13 @@ class PublicSigner implements SignerInterface
      */
     public static function calculateKeyId(PublicKey $publicKey)
     {
-        return Base64UrlSafe::encodeUnpadded(
+        return rtrim(Base64UrlSafe::encode(
             hash(
                 'sha256',
                 $publicKey->raw(),
                 true
             )
-        );
+        ), '=');
     }
 
     /**
