diff --git a/src/Jwt.php b/src/Jwt.php
index 01249cf..232815d 100644
--- a/src/Jwt.php
+++ b/src/Jwt.php
@@ -79,9 +79,9 @@ abstract class Jwt
             $headerData['kid'] = $this->keyId;
         }
 
-        $jwtHeader = Base64UrlSafe::encodeUnpadded(Json::encode($headerData));
-        $jwtPayload = Base64UrlSafe::encodeUnpadded(Json::encode($jsonData));
-        $jwtSignature = Base64UrlSafe::encodeUnpadded($this->sign($jwtHeader.'.'.$jwtPayload));
+        $jwtHeader = \rtrim(Base64UrlSafe::encode(Json::encode($headerData)), '=');
+        $jwtPayload = \rtrim(Base64UrlSafe::encode(Json::encode($jsonData)), '=');
+        $jwtSignature = \rtrim(Base64UrlSafe::encode($this->sign($jwtHeader.'.'.$jwtPayload)), '=');
 
         return $jwtHeader.'.'.$jwtPayload.'.'.$jwtSignature;
     }
diff --git a/src/Keys/EdDSA/PublicKey.php b/src/Keys/EdDSA/PublicKey.php
index bf3b9ee..37892dd 100644
--- a/src/Keys/EdDSA/PublicKey.php
+++ b/src/Keys/EdDSA/PublicKey.php
@@ -49,7 +49,7 @@ class PublicKey
      */
     public function encode()
     {
-        return Base64UrlSafe::encodeUnpadded($this->publicKey);
+        return \rtrim(Base64UrlSafe::encode($this->publicKey), '=');
     }
 
     /**
diff --git a/src/Keys/EdDSA/SecretKey.php b/src/Keys/EdDSA/SecretKey.php
index 7718935..8306cb3 100644
--- a/src/Keys/EdDSA/SecretKey.php
+++ b/src/Keys/EdDSA/SecretKey.php
@@ -69,7 +69,7 @@ class SecretKey
      */
     public function encode()
     {
-        return Base64UrlSafe::encodeUnpadded($this->secretKey);
+        return \rtrim(Base64UrlSafe::encode($this->secretKey), '=');
     }
 
     /**
diff --git a/src/Keys/HS256/SecretKey.php b/src/Keys/HS256/SecretKey.php
index fbdab57..756d92b 100644
--- a/src/Keys/HS256/SecretKey.php
+++ b/src/Keys/HS256/SecretKey.php
@@ -60,7 +60,7 @@ class SecretKey
      */
     public function encode()
     {
-        return Base64UrlSafe::encodeUnpadded($this->secretKey);
+        return \rtrim(Base64UrlSafe::encode($this->secretKey), '=');
     }
 
     /**
diff --git a/tests/EdDSATest.php b/tests/EdDSATest.php
index 13d2dc9..dd4ca8a 100644
--- a/tests/EdDSATest.php
+++ b/tests/EdDSATest.php
@@ -83,7 +83,7 @@ class EdDSATest extends TestCase
             $secretKey->getPublicKey(),
             $secretKey
         );
-        $this->assertSame('hgyY0il_MGCjP0JzlnLWG1PPOt7-09PGcvMg3AIbQR6dWbhijcNR4ki4iylGjg5BhVsPt9g7sVvpAr_MuM0KAg', Base64UrlSafe::encodeUnpadded($dsa->sign('eyJhbGciOiJFZERTQSJ9.RXhhbXBsZSBvZiBFZDI1NTE5IHNpZ25pbmc')));
+        $this->assertSame('hgyY0il_MGCjP0JzlnLWG1PPOt7-09PGcvMg3AIbQR6dWbhijcNR4ki4iylGjg5BhVsPt9g7sVvpAr_MuM0KAg', \rtrim(Base64UrlSafe::encode($dsa->sign('eyJhbGciOiJFZERTQSJ9.RXhhbXBsZSBvZiBFZDI1NTE5IHNpZ25pbmc')), '='));
     }
 
     public function testTestVectorVerify()
